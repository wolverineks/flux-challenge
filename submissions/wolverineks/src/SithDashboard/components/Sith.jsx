// Sith.jsx
// @flow

import React, {
  Component
} from 'react'

import type {
  ClassName,
  Planet,
  Name
} from '../../Types'

type Props = {
  className: ClassName,
  name: Name | void,
  homeworld: Planet | void
}
type State = {}

export default class Sith extends Component<Props, State> {
  render () {
    const {
      className,
      name,
      homeworld
    } = this.props
    const homeworldText = homeworld
      ? 'Homeworld: ' + homeworld.name
      : undefined

    return <li className={className}>
      <h3>{name}</h3>
      <h6>{homeworldText}</h6>
    </li>
  }
}
