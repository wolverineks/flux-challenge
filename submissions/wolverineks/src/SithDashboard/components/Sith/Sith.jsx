// SithDashboard/components/Sith.jsx
// @flow

import React, {Component} from 'react'

import {HOMEWORLD_TEXT} from './strings'

import type {Planet, Name} from './Types'
type Props = {name: ?Name, homeworld: ?Planet}
type State = {}

export default class Sith extends Component<Props, State> {
  render () {
    const {name, homeworld} = this.props

    const homeworldText = homeworld
      ? HOMEWORLD_TEXT + homeworld.name
      : undefined

    return <div>
      <h3>{name}</h3>
      <h6>{homeworldText}</h6>
    </div>
  }
}
