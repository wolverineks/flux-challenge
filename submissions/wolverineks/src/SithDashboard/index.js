// SithDashboard/index.js
// @flow

import selectors from './selectors'
import reducer from './reducer'

import Main from './SithDashboard.jsx'
import connector from './connector'

export {
  Main,
  reducer,
  selectors,
}

export default connector
