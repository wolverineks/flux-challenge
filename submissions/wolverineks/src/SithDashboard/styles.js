// SithDashboard/styles.js
// @flow

export const classNames = {
  main: 'app-container',
  mainInner: 'css-root',
  currentPlanetIndicator: 'css-planet-monitor',
  scrollableList: 'css-scrollable-list',
  sithList: 'css-slots',
  sith: 'css-slot',
  scrollButtons: 'css-scroll-buttons',
  buttonUp: 'css-button-up',
  buttonDown: 'css-button-down'
}
