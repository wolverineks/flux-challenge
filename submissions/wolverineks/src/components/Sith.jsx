// @flow
import React, {
  Component
} from 'react'

import type * as Types from '../Types'
type Props = {
  className: Types.ClassName,
  homeworld: Types.Planet,
  name: Types.Name
}
type State = {}

export default class Sith extends Component<Props, State> {
  render () {
    const {className, name, homeworld} = this.props
    const homeworldText = `Homeworld: ${homeworld.name}`
    return <li className={className}>
      <h3>{name}</h3>
      <h6>{homeworldText}</h6>
    </li>
  }
}
