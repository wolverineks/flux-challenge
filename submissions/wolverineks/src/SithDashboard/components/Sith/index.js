// @flow

import Sith from './Sith'

export {
  Sith
}
export default Sith
