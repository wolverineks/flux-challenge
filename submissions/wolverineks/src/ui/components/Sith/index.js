// @flow

import Sith from './Sith'
import * as strings from './strings'
import type {Planet, Name, LastUpdated} from '../../../state/types'

export type {Planet, Name, LastUpdated}
export {
  strings,
  Sith
}
export default Sith
