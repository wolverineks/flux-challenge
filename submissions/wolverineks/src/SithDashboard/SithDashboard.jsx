// SithDashboard/SithDashboard.jsx
// @flow

import React, {Component} from 'react'

import {Button, CurrentPlanetIndicator, ScrollableList, List, Item, ScrollButtons, Sith} from './components'
import {classNames} from './styles'

import {type Planet, type Siths} from './Types'
type Props = {planet: Planet, siths: Siths}
type State = {}

export default class SithDashboard extends Component<Props, State> {
  render () {
    const {planet, siths} = this.props

    return <div className={classNames.main}>
      <div className={classNames.mainInner}>

        <CurrentPlanetIndicator planet={planet} />

        <ScrollableList>
          <List>
            <Item><Sith name={siths[0] ? siths[0].name : undefined} homeworld={siths[0] ? siths[0].homeworld : undefined} /></Item>
            <Item><Sith name={siths[1] ? siths[1].name : undefined} homeworld={siths[1] ? siths[1].homeworld : undefined} /></Item>
            <Item><Sith name={siths[2] ? siths[2].name : undefined} homeworld={siths[2] ? siths[2].homeworld : undefined} /></Item>
            <Item><Sith name={siths[3] ? siths[3].name : undefined} homeworld={siths[3] ? siths[3].homeworld : undefined} /></Item>
            <Item><Sith name={siths[4] ? siths[4].name : undefined} homeworld={siths[4] ? siths[4].homeworld : undefined} /></Item>
          </List>

          <ScrollButtons>
            <Button direction={'up'} />
            <Button direction={'down'} />
          </ScrollButtons>
        </ScrollableList>

      </div>
    </div>
  }
}
