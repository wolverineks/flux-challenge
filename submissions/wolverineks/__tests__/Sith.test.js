import React from 'react'
import renderer from 'react-test-renderer'
import Sith from '../src/SithDashboard/components/Sith'
import {classNames} from '../src/SithDashboard/styles'

describe('Sith', () => {
  describe('When has name', () => {
    describe('When has planet', () => {
      test('Sith', () => {
        const className = classNames.sith
        const name = 'Skere Kaan'
        const planet = {id: 1, name: 'Coruscant'}
        const component = renderer.create(<Sith name={name} homeworld={planet} className={className}/>)
        const tree = component.toJSON()
        expect(tree).toMatchSnapshot()
      })
    })

    describe('When lacks planet', () => {
      test('Sith', () => {
        const className = classNames.sith
        const name = 'Skere Kaan'
        const planet = undefined
        const component = renderer.create(<Sith name={name} homeworld={planet} className={className}/>)
        const tree = component.toJSON()
        expect(tree).toMatchSnapshot()
      })
    })
  })

  describe('When lacks name', () => {
    describe('When has planet', () => {
      test('Sith', () => {
        const className = classNames.sith
        const name = undefined
        const planet = {id: 1, name: 'Coruscant'}
        const component = renderer.create(<Sith name={name} homeworld={planet} className={className}/>)
        const tree = component.toJSON()
        expect(tree).toMatchSnapshot()
      })
    })

    describe('When lacks planet', () => {
      test('Sith', () => {
        const className = classNames.sith
        const name = undefined
        const planet = undefined
        const component = renderer.create(<Sith name={name} homeworld={planet} className={className}/>)
        const tree = component.toJSON()
        expect(tree).toMatchSnapshot()
      })
    })
  })
})
