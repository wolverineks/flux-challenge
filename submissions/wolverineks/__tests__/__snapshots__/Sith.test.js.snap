// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sith When has name When has planet Sith 1`] = `
<li
  className="css-slot"
>
  <h3>
    Skere Kaan
  </h3>
  <h6>
    Homeworld: Coruscant
  </h6>
</li>
`;

exports[`Sith When has name When lacks planet Sith 1`] = `
<li
  className="css-slot"
>
  <h3>
    Skere Kaan
  </h3>
  <h6 />
</li>
`;

exports[`Sith When lacks name When has planet Sith 1`] = `
<li
  className="css-slot"
>
  <h3 />
  <h6>
    Homeworld: Coruscant
  </h6>
</li>
`;

exports[`Sith When lacks name When lacks planet Sith 1`] = `
<li
  className="css-slot"
>
  <h3 />
  <h6 />
</li>
`;
